src/v/cluster/partition.h

```diff
@@ -343,10 +344,13 @@ class partition : public ss::enable_lw_shared_from_this<partition> {
     ss::shared_ptr<cloud_storage::async_manifest_view>
     get_cloud_storage_manifest_view();
 
-    ss::future<std::error_code> disable_writes();
-    ss::future<std::error_code> enable_writes();
+    ss::future<std::error_code> toggle_writes(
```

Comment:
nit: maybe something like `set_writes_disabled` will be a better name ? A toggle suggest that the function switches the value from one to the other